components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/client/IdentityProviderMgtServiceClient.java

```diff
@@ -113,22 +122,116 @@ public void updateResidentIdP(IdentityProvider identityProvider) throws Exceptio
      *
      * @return List of <code>FederatedIdentityProvider</code>. IdP names, primary IdP and home realm
      * identifiers of each IdP
-     * @throws Exception Error when getting list of Identity Providers
+     * @throws IdentityException Error when getting list of Identity Providers
      */
-    public List<IdentityProvider> getIdPs() throws Exception {
+    public IdentityProvider[] getIdPs() throws IdentityException {
+
         try {
-            IdentityProvider[] identityProviders = idPMgtStub.getAllIdPs();
-            if (identityProviders != null && identityProviders.length > 0) {
-                return Arrays.asList(identityProviders);
-            } else {
-                return new ArrayList<IdentityProvider>();
-            }
-        } catch (Exception e) {
+            return idPMgtStub.getAllIdPs();
+        } catch (RemoteException | IdentityProviderMgtServiceIdentityProviderManagementExceptionException e) {
             log.error("Error in retrieving the list of Identity Providers for a given tenant", e);
-            throw new Exception("Error occurred while retrieving list of Identity Providers");
+            throw new IdentityException("Error occurred while retrieving list of Identity Providers", e);
+        }
+    }
+
+    /**
+     * Get all list of Idp information with pagination.
+     *
+     * @param pageNumber pageNumber need to filter.
+     * @return list of paginated idp array.
+     * @throws Exception Error when getting list of Identity Providers.
+     */
+    public IdentityProvider[] getAllPaginatedIdpInfo(int pageNumber) throws IdentityException {
+
+        try {
+            return idPMgtStub.getAllPaginatedIdpInfo(pageNumber);
+        } catch (RemoteException | IdentityProviderMgtServiceIdentityProviderManagementExceptionException e) {
+            throw new IdentityException("Error in retrieving the list of Identity Providers for a given tenant", e);
```

Comment:
as this method is directly called from UI, let's avoid adding 'e' to the error since it will show in the error message in UI
log the error before throwing